### PR TITLE
 Add abandoned cart status page

### PR DESCRIPTION
**Pull request by TomAbel**
for issue #65

### CODE REVIEW

File: upload/admin/view/template/extension/module/smaily_for_opencart.tpl [hunk 1]
```diff
@@ -392,20 +392,69 @@
...
+                  <thead>
+                    <tr>
+                      <th scope="col"><?php echo $cart_status_table_header_id; ?></th>
+                      <th class="text-left"><a href="<?php echo $cart_status_table_sort_name_link; ?>" class="<?php echo $sort ==='lastname' ? 'strtolower($order)' : '' ?>"><?php echo $cart_status_table_header_name; ?></a></th>
+                      <th class="text-left"><a href="<?php echo $cart_status_table_sort_email_link; ?>" class="<?php echo $sort ==='email' ? 'strtolower($order)' : '' ?>"><?php echo $cart_status_table_header_email; ?></a></th>
+                      <th class="text-left"><?php echo $cart_status_table_header_cart; ?></th>
+                      <th class="text-left"><a href="<?php echo $cart_status_table_sort_date_link; ?>" class="<?php echo $sort ==='sent_time' ? 'strtolower($order)' : '' ?>"><?php echo $cart_status_table_header_date; ?></a></th>
+                      <th class="text-left"><a href="<?php echo $cart_status_table_sort_status_link; ?>" class="<?php echo $sort ==='is_sent' ? 'strtolower($order)' : '' ?>"><?php echo $cart_status_table_header_status; ?></a></th>
```
Comment (kaittodesk) [inline upload/admin/view/template/extension/module/smaily_for_opencart.tpl:405]: Formatting: missing space after condition operator.

```suggestion
                      <th class="text-left"><a href="<?php echo $cart_status_table_sort_name_link; ?>" class="<?php echo $sort === 'lastname' ? 'strtolower($order)' : '' ?>"><?php echo $cart_status_table_header_name; ?></a></th>
                      <th class="text-left"><a href="<?php echo $cart_status_table_sort_email_link; ?>" class="<?php echo $sort === 'email' ? 'strtolower($order)' : '' ?>"><?php echo $cart_status_table_header_email; ?></a></th>
                      <th class="text-left"><?php echo $cart_status_table_header_cart; ?></th>
                      <th class="text-left"><a href="<?php echo $cart_status_table_sort_date_link; ?>" class="<?php echo $sort === 'sent_time' ? 'strtolower($order)' : '' ?>"><?php echo $cart_status_table_header_date; ?></a></th>
                      <th class="text-left"><a href="<?php echo $cart_status_table_sort_status_link; ?>" class="<?php echo $sort === 'is_sent' ? 'strtolower($order)' : '' ?>"><?php echo $cart_status_table_header_status; ?></a></th>
```

File: upload/admin/view/template/extension/module/smaily_for_opencart.tpl [hunk 2]
```diff
@@ -392,20 +392,69 @@
...
+                    </tr>
+                  </thead>
+                  <tbody>
+                    <?php foreach($abandoned_cart_list as $abandoned_cart): ?>
+                    <tr>
+                      <th scope="row"><?php echo $abandoned_cart['customer_id']; ?></th>
+                        <td><?php echo $abandoned_cart['firstname'] . $abandoned_cart['lastname']; ?></td>
```
Comment (kaittodesk) [inline upload/admin/view/template/extension/module/smaily_for_opencart.tpl:412]: Missing space between first and last name.

File: upload/admin/view/template/extension/module/smaily_for_opencart.tpl [hunk 3]
```diff
@@ -392,20 +392,69 @@
...
+                      <th class="text-left"><a href="<?php echo $cart_status_table_sort_email_link; ?>" class="<?php echo $sort ==='email' ? 'strtolower($order)' : '' ?>"><?php echo $cart_status_table_header_email; ?></a></th>
+                      <th class="text-left"><?php echo $cart_status_table_header_cart; ?></th>
+                      <th class="text-left"><a href="<?php echo $cart_status_table_sort_date_link; ?>" class="<?php echo $sort ==='sent_time' ? 'strtolower($order)' : '' ?>"><?php echo $cart_status_table_header_date; ?></a></th>
+                      <th class="text-left"><a href="<?php echo $cart_status_table_sort_status_link; ?>" class="<?php echo $sort ==='is_sent' ? 'strtolower($order)' : '' ?>"><?php echo $cart_status_table_header_status; ?></a></th>
+                    </tr>
+                  </thead>
+                  <tbody>
+                    <?php foreach($abandoned_cart_list as $abandoned_cart): ?>
+                    <tr>
+                      <th scope="row"><?php echo $abandoned_cart['customer_id']; ?></th>
```
Comment (kaittodesk) [inline upload/admin/view/template/extension/module/smaily_for_opencart.tpl:411]: Customer ID should be wrapped with `<td>` not `<th>`.

File: upload/admin/view/template/extension/module/smaily_for_opencart.tpl [hunk 4]
```diff
@@ -392,20 +392,69 @@
...
+                    </tr>
+                    <?php endforeach; ?>
+                  </tbody>
+                </table>
+                <div class="row">
+                  <div class="col-sm-6 text-left"><?php echo $pagination; ?></div>
+                  <div class="col-sm-6 text-right"><?php echo $results; ?></div>
+                </div>
+              </div>
+            </div>
```
Comment (kaittodesk) [inline upload/admin/view/template/extension/module/smaily_for_opencart.tpl:438]: Formatting: indentation.

File: upload/admin/view/template/extension/module/smaily_for_opencart.tpl [hunk 5]
```diff
@@ -597,6 +646,18 @@
...
+    var query_parameters = window.location.href.slice(window.location.href.indexOf('?') + 1).split('&');
+    var i;
+    for (i = 0; i < query_parameters.length; i++) {
+      parameter = query_parameters[i].split('=')[0];
+      if ($.inArray(parameter, search_params) !== -1) {
+        return true;
+      }
+    }
+    return false;
+  }
```
Comment (kaittodesk) [inline upload/admin/view/template/extension/module/smaily_for_opencart.tpl:660]: Formatting: indentation.

**Review by kaittodesk — CHANGES_REQUESTED** [review]

File: upload/admin/view/template/extension/module/smaily_for_opencart.tpl (hunk 4 above)
Comment (TomAbel) [reply upload/admin/view/template/extension/module/smaily_for_opencart.tpl:438]: I don't see it right away. You mean this?
```
                       <td scope="row"><?php echo $abandoned_cart['customer_id']; ?></td>
-                        <td><?php echo $abandoned_cart['firstname'] . " " . $abandoned_cart['lastname']; ?></td>
-                        <td><?php echo $abandoned_cart['email']; ?></td>
+                      <td><?php echo $abandoned_cart['firstname'] . " " . $abandoned_cart['lastname']; ?></td>
+                      <td><?php echo $abandoned_cart['email']; ?></td>
```

Comment (kaittodesk) [reply upload/admin/view/template/extension/module/smaily_for_opencart.tpl:438]: I guess so. Now it seems to be fixed.

File: upload/admin/view/template/extension/module/smaily_for_opencart.tpl [hunk 6]
```diff
@@ -392,20 +392,69 @@
...
+                              <td width="70%"><a href="<?php echo $product_url_without_id . $product['data']['product_id']; ?>"><?php echo $product['data']['name']; ?></a></td>
+                              <td width="15%"><?php echo "x " . $product['quantity']; ?></td>
+                              <td width="15%"><?php echo "$" . $product['data']['price']; ?></td>
+                            </tr>
+                            <?php endforeach; ?>
+                          </tbody>
+                        </table>
+                      </td>
+                      <td><?php echo $abandoned_cart['sent_time'] ?: ''; ?><br></td>
+                      <td><b><?php echo $abandoned_cart['is_sent'] == '1' ? 'SENT' : 'PENDING'; ?><b></td>
```
Comment (kaittodesk) [inline upload/admin/view/template/extension/module/smaily_for_opencart.tpl:428]: Discovered there is an unclosed `<b>` tag:

```suggestion
                      <td><b><?php echo $abandoned_cart['is_sent'] == '1' ? 'SENT' : 'PENDING'; ?></b></td>
```

**Review by kaittodesk — CHANGES_REQUESTED** [review]

**Review by kaittodesk — APPROVED** [review]

**Pull request merged**